### PR TITLE
[ROCm] Turn off unaligned access mode

### PR DESCRIPTION
**Pull request by qedawkins**
There are correctness issues related to unaligned accesses. See https://github.com/iree-org/iree/issues/18798

### CODE REVIEW

File: compiler/plugins/target/ROCM/ROCMTarget.cpp [hunk 1]
```diff
@@ -473,9 +473,13 @@ class ROCMTargetBackend final : public TargetBackend {
         // https://github.com/iree-org/iree/issues/18746 so we disable this
         // feature.
         if (targetArch.starts_with("gfx9")) {
-          features.emplace_back("-fma-mix-insts");
+          features.emplace_back("-fma-mix-insts,-unaligned-access-mode");
```
Comment (nirvedhmeshram) [inline compiler/plugins/target/ROCM/ROCMTarget.cpp]: This one is not needed?

**Comment by nirvedhmeshram** [comment]
Ouch, sdxl is very slow with this

https://github.com/iree-org/iree/actions/runs/11584595911/job/32252523388#step:8:144

**Comment by qedawkins** [comment]
😿

**Comment by MaheshRavishankar** [comment]
Lets just turn on SLP vectorizer and call it a day?

**Comment by qedawkins** [comment]
That's the hope now. Nirvedh is doing a CI run and checking perf I think.

**Comment by qedawkins** [comment]
Closing because I don't think we want this.

**Pull request closed without merging**